BackendNew/controllers/students.js

```diff
@@ -263,3 +262,38 @@ const updateStudentSelectCompanyInfo = (req, isUpdated) => {
   });
   isUpdated(null, { message: "updated" });
 };
+
+// Remove student.
+exports.removeStudent = (req, res) => {
+  var batch = req.query.batch;
+  if (batch) {
+    var studentId = req.params.studentId;
+
+    students.removeStudentHasBatch(studentId, batch, (err, result) => {
+      if (err) {
+        res.status(500).send();
+      } else {
+        // If no one is removed
+        if (result.data.affectedRows == 0) {
+          res
+            .status(400)
+            .send({ message: "There is no student with the batch" });
+        } else {
+          students.removeStudent(studentId, (err, result) => {
+            if (err) {
+              res.status(400).send({ message: "Cannot removed the student" });
```

Comment:
Will this rollback previous changes done to  table " StudentHasBatch"?

Reply:
No. should we do that ?

Reply:
If error occurred at this stage there will be student who does not have a batch. Wont't this affect badly at code later?

Reply:
What do you suggest for this?